### DISCUSSION TITLE
How would I run Dependency Track behind a Nginx on a single domain

### DISCUSSION THREAD

**Discussion by andrepaulotek** [DependencyTrack/dependency-track#3355] [Q&A]
I tried running Dependency Track locally on 2 different domains (dependency.local and api.dependency.local).
Because I'm using local domains, the certificates that I'm using are self-signed and there's a big warning before opening it the first time. The problem is once you accept the self-signed certificate for dependency.local, its not accepted for api.dependency.local and it will fail the login.

So I tried to switch everything to a single domain and move the API to dependency.local/api but doesn't seem to work.
This is my config:
```
server {
  listen 443 ssl http2;
  listen [::]:443 ssl http2;

  server_name dependency.local;

  include snippets/self-signed.conf;
  include snippets/ssl-params.conf;

  location / {
    proxy_pass http://127.0.0.1:9102;
    proxy_read_timeout 90;
    proxy_connect_timeout 90;
    proxy_set_header Host $http_host;
    proxy_set_header X-Forwarded-Host $host;
    proxy_set_header X-Forwarded-Server $host;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
  }

  location /api/ {
    proxy_pass http://127.0.0.1:9101;
    proxy_read_timeout 90;
    proxy_connect_timeout 90;
    proxy_set_header Host $http_host;
    proxy_set_header X-Forwarded-Host $host;
    proxy_set_header X-Forwarded-Server $host;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
  }

}

```

If I visit https://dependency.local/api I see the error message: `Uncaught internal server error`
Is it possible to run this on a single domain on separate subdirectories?

**Comment by psevestre** [comment]
Yes, it is possible. I'm running the API and UI containers as two services in Kubernetes. The ingress setup looks very similar to your ngnix setup. I'm no NGNIX experts, but maybe the issue has something to do with the priority between both "location" elements. Since "/api" also matches "/", you must ensure it is evaluated first.
